### PR TITLE
unbound: 1.19.2 pkg-config for protobuf-c

### PR DESCRIPTION
**Pull request by haraldh**
## Description of changes

Patched configure script with:
https://github.com/NLnetLabs/unbound/commit/59d98b9ef64e597c331c27160715d7a1b40c8638 Credits: @NickCao

Enables multiple outputs for protobuf-c as in https://github.com/NixOS/nixpkgs/pull/283783

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by vcunat — CHANGES_REQUESTED** [review]
When I apply this to master and set `withDNSTAP ? true`, I get a failure
```
checking for library containing protobuf_c_message_pack... no
configure: error: The protobuf-c library was not found. Please install the development libraries for protobuf-c!
```

**Comment by NickCao** [comment]
There was a followup: https://github.com/NLnetLabs/unbound/issues/1006

**Comment by SuperSandro2000** [comment]
I think the fix is https://github.com/NLnetLabs/unbound/commit/3f5175584b0bb9ff7d417bc195ec6e4316ae58d3

**Comment by haraldh** [comment]
added both followup patches

**Comment by haraldh** [comment]
hmm.. ofborg-eval seems to be broken after rebasing to current staging

**Comment by SuperSandro2000** [comment]
@ofborg build unbound

File: pkgs/tools/networking/unbound/default.nix [hunk 1]
```diff
@@ -58,6 +58,29 @@ stdenv.mkDerivation (finalAttrs: {
...
+    # Search for protobuf-c with pkg-config
+    # https://github.com/NLnetLabs/unbound/pull/999
+    (fetchpatch {
+      url = "https://github.com/NLnetLabs/unbound/commit/59d98b9ef64e597c331c27160715d7a1b40c8638.patch";
+      hash = "sha256-DvYoYBTXOwbR8Z0GRgt724WqX3dbIEOdICU2/VMRSVQ=";
+    })
+    # Fix for previous patch
+    # https://github.com/NLnetLabs/unbound/issues/1006
+    (fetchpatch {
+      url = "https://github.com/NLnetLabs/unbound/commit/93490a0fc1bf9e62e6edcd6b69f1463c7ac410e9.patch";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/unbound/default.nix:73]:
```suggestion
      url = "https://github.com/NLnetLabs/unbound/commit/93490a0fc1bf9e62e6edcd6b69f1463c7ac410e9.patch";
      excludes = [ "doc/Changelog" ];
```

```
building '/nix/store/bhk94rq5bxgva0h97p8yaaqvp5va1k2m-unbound-1.19.1.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/dah45yk4pmsxgvs0fbzhaq1g25h08m4y-unbound-1.19.1.tar.gz
source root is unbound-1.19.1
setting SOURCE_DATE_EPOCH to timestamp 1707825929 of file unbound-1.19.1/util/configparser.c
Running phase: patchPhase
applying patch /nix/store/8maj3xnkf65v47iqcm3ijy7acnpxlf1y-59d98b9ef64e597c331c27160715d7a1b40c8638.patch
patching file dnstap/dnstap.m4
applying patch /nix/store/7irj3hhkccx3x7r7zdk4aggkzfg0kflm-93490a0fc1bf9e62e6edcd6b69f1463c7ac410e9.patch
patching file configure.ac
patching file dnstap/dnstap.m4
patching file doc/Changelog
Hunk #1 FAILED at 1.
1 out of 1 hunk FAILED -- saving rejects to file doc/Changelog.rej
error: builder for '/nix/store/bhk94rq5bxgva0h97p8yaaqvp5va1k2m-unbound-1.19.1.drv' failed with exit code 1;
```

also please update the hash

**Comment by vcunat** [comment]
The eval problem (`llvmPackages.mlir`) has been fixed.

**Comment by NickCao** [comment]
@ofborg eval

File: pkgs/tools/networking/unbound/default.nix (hunk 1 above)
Comment (haraldh) [reply pkgs/tools/networking/unbound/default.nix:73]: sorry... done and checked compilation on master branch

**Comment by haraldh** [comment]
```
error: building of '/nix/store/j1ylkl7kc54pgrxpl9mlyzh3x10k92wk-llvm-16.0.6.drv^dev' from .drv file timed out after 3600 seconds
```

**Comment by haraldh** [comment]
rebased on current staging

**Comment by vcunat** [comment]
Maybe update to .2 instead of applying patches?  Notably I see in announcement:
>  Fix CVE-2024-1931, Denial of service when trimming EDE text on positive replies.

**Comment by SuperSandro2000** [comment]
Works on linux, darwin we can only reasonable test in staging-next, anyway

**Pull request merged**